src/main/java/net/neoforged/neoforge/client/textures/FluidSpriteCache.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.textures;
+
+import java.util.Map;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.texture.MissingTextureAtlasSprite;
+import net.minecraft.client.renderer.texture.TextureAtlas;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.core.BlockPos;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.level.BlockAndTintGetter;
+import net.minecraft.world.level.material.FluidState;
+import net.neoforged.neoforge.client.extensions.common.IClientFluidTypeExtensions;
+
+public final class FluidSpriteCache {
```

Comment:
Should this class be marked as `@ApiStatus.Internal`, or at the very least the `reload` method?

Reply:
I intentionally didn't mark the class as internal because I can see mods making use of this as well. Marking `reload()` as internal seems reasonable though, I'll change that shortly.